chainntnfs/test_utils.go

```diff
@@ -170,6 +171,17 @@ func NewMiner(t *testing.T, netParams *chaincfg.Params, extraArgs []string,
 
 	t.Helper()
 
+	// Go starts a new sub process for each test package, but we want to
+	// have unique listening addresses across all packages in case things
+	// run parallel. That's why we use the parent process id to generate
+	// unique addresses.
+	parentProcess := os.Getppid()
```

Comment:
Feel like this can just be combined with `lntest.NewMiner` with some modifications, guess we can clean it up in future PRs.